packages/iov-ethereum/src/ethereumcodec.ts

```diff
@@ -30,24 +34,64 @@ import {
   encodeQuantity,
   fromBcpChainId,
   normalizeHex,
+  shouldBeInterpretedAsErc20Transfer,
 } from "./utils";
 
-export const ethereumCodec: TxCodec = {
-  bytesToSign: (unsigned: UnsignedTransaction, nonce: Nonce): SigningJob => {
+/**
+ * See https://github.com/ethereum/wiki/wiki/JSON-RPC#eth_gettransactionbyhash
+ *
+ * This interface is package-internal.
+ */
+export interface EthereumRpcTransactionResult {
+  readonly blockHash: string;
+  readonly blockNumber: string;
+  readonly from: string;
+  /** Gas limit as set by the user */
+  readonly gas: string;
+  readonly gasPrice: string;
+  readonly hash: string;
+  readonly input: string;
+  readonly nonce: string;
+  readonly r: string;
+  readonly s: string;
+  readonly to: string;
+  readonly transactionIndex: string;
+  readonly v: string;
+  readonly value: string;
+}
+
+export interface EthereumCodecOptions {
+  /** List of supported ERC20 tokens */
```

Comment:
"List" is a little confusing. Also what does "supported" mean in this context? (Supported by whom/what?)

Reply:
```
  /**
   * ERC20 tokens supported by the codec instance.
   *
   * The behaviour of encoding/decoding transactions for other tokens is undefined.
   */
```